deployer/src/deployment/queue.rs

```diff
@@ -117,53 +117,52 @@ fn extract_tar_gz_data(data: impl Read, dest: impl AsRef<Path>) -> Result<()> {
     Ok(())
 }
 
-/// Check for an existing marker file in the specified project directory and
-/// if one exists delete the indicated '.so' file.
-async fn remove_old_build(project_path: impl AsRef<Path>) -> Result<()> {
-    let marker_path = project_path.as_ref().join(MARKER_FILE_NAME);
-
-    if let Ok(mut existing_marker_file) = fs::File::open(&marker_path).await {
-        let mut old_so_name = String::new();
-        existing_marker_file
-            .read_to_string(&mut old_so_name)
-            .await?;
-
-        let old_so_path = project_path.as_ref().join(old_so_name);
-
-        if old_so_path.exists() {
-            fs::remove_file(old_so_path).await?;
+/// Check for a '.so' file specified in the database for the given deployment
+/// name and, if one is found, delete it from the libs directory.
+async fn remove_old_lib(
+    persistence: &Persistence,
+    name: &str,
+    libs_path: impl AsRef<Path>,
+) -> Result<()> {
+    if let Some(lib_name) = persistence.get_last_built_lib(name).await? {
+        let lib_path = libs_path.as_ref().join(lib_name);
+
+        if lib_path.exists() {
+            fs::remove_file(lib_path).await?;
```

Comment:
I'm wondering, what will happen when a new deployment version fails to start during run. We will want to return to a previous instance, right?

Maybe we shouldn't delete old 'so' files?

Reply:
@chesedo Very point - perhaps the deletion logic should be moved into the loading/execution stage then?